x-pack/plugins/observability/server/lib/rules/slo_burn_rate/executor.test.ts

```diff
@@ -0,0 +1,279 @@
+} from '@kbn/core/server/mocks';
+import { LifecycleAlertService, LifecycleAlertServices } from '@kbn/rule-registry-plugin/server';
+import { PublicAlertFactory } from '@kbn/alerting-plugin/server/alert/create_alert_factory';
+import { ISearchStartSearchSource } from '@kbn/data-plugin/public';
+import { MockedLogger } from '@kbn/logging-mocks';
+import { SanitizedRuleConfig } from '@kbn/alerting-plugin/common';
+import { Alert, RuleExecutorServices } from '@kbn/alerting-plugin/server';
+import {
+  ALERT_EVALUATION_THRESHOLD,
+  ALERT_EVALUATION_VALUE,
+  ALERT_REASON,
+} from '@kbn/rule-data-utils';
+import {
+  BurnRateAlertContext,
+  BurnRateAlertState,
+  BurnRateAllowedActionGroups,
+  BurnRateRuleParams,
+  FIRED_ACTION,
+  AlertStates,
+  getRuleExecutor,
+} from './executor';
+import { aStoredSLO, createSLO } from '../../../services/slo/fixtures/slo';
+import { SLO } from '../../../domain/models';
+
+const commonEsResponse = {
+  took: 100,
+  timed_out: false,
+  _shards: {
+    total: 0,
+    successful: 0,
+    skipped: 0,
+    failed: 0,
+  },
+  hits: {
+    hits: [],
+  },
+};
+
+const BURN_RATE_THRESHOLD = 2;
+const BURN_RATE_ABOVE_THRESHOLD = BURN_RATE_THRESHOLD + 0.01;
+const BURN_RATE_BELOW_THRESHOLD = BURN_RATE_THRESHOLD - 0.01;
+
+describe('BurnRateRuleExecutor', () => {
+  let esClientMock: ElasticsearchClientMock;
+  let soClientMock: jest.Mocked<SavedObjectsClientContract>;
+  let loggerMock: jest.Mocked<MockedLogger>;
+  let alertWithLifecycleMock: jest.MockedFn<LifecycleAlertService>;
+  let alertFactoryMock: jest.Mocked<
+    PublicAlertFactory<BurnRateAlertState, BurnRateAlertContext, BurnRateAllowedActionGroups>
+  >;
+  let searchSourceClientMock: jest.Mocked<ISearchStartSearchSource>;
+  let uiSettingsClientMock: jest.Mocked<IUiSettingsClient>;
+  let servicesMock: RuleExecutorServices<
+    BurnRateAlertState,
+    BurnRateAlertContext,
+    BurnRateAllowedActionGroups
+  > &
+    LifecycleAlertServices<BurnRateAlertState, BurnRateAlertContext, BurnRateAllowedActionGroups>;
+
+  beforeEach(() => {
+    esClientMock = elasticsearchServiceMock.createElasticsearchClient();
+    soClientMock = savedObjectsClientMock.create();
+    alertWithLifecycleMock = jest.fn();
+    alertFactoryMock = {
+      create: jest.fn(),
+      done: jest.fn(),
+      alertLimit: { getValue: jest.fn(), setLimitReached: jest.fn() },
+    };
+    loggerMock = loggingSystemMock.createLogger();
+    servicesMock = {
+      alertWithLifecycle: alertWithLifecycleMock,
+      savedObjectsClient: soClientMock,
+      scopedClusterClient: { asCurrentUser: esClientMock, asInternalUser: esClientMock },
+      alertFactory: alertFactoryMock,
+      searchSourceClient: searchSourceClientMock,
+      uiSettingsClient: uiSettingsClientMock,
+      shouldWriteAlerts: jest.fn(),
+      shouldStopExecution: jest.fn(),
+      getAlertStartedDate: jest.fn(),
+      getAlertUuid: jest.fn(),
+      getAlertByAlertUuid: jest.fn(),
+    };
+  });
```

Comment:
💬 The boilerplate is heavy, but the important ones are esClient, soClient, alertWithLifecycle and alertFactory